quickfixj-core/src/main/java/quickfix/mina/initiator/IoSessionInitiator.java

```diff
@@ -261,8 +261,10 @@ private void handleConnectException(Throwable e) {
             final String nextRetryMsg = " (Next retry in " + computeNextRetryConnectDelay() + " milliseconds)";
             if (e instanceof IOException) {
                 fixSession.getLog().onErrorEvent(e.getClass().getName() + " during connection to " + socketAddress + ": " + e + nextRetryMsg);
+                fixSession.getStateListener().onConnectException((IOException) e);
```

Comment:
Little smelly to expose and call another classes listeners (Law of Demeter) but not egregiously so.

Reply:
I agree. What's the alternative? Pass the listener through the Initiator implementations or create an all-new Initiator exception listener?

Reply:
The line above already sets precedent for calling through `Session.getLog()` and the usecase appears (since I reported it after many years) "edge" so I wouldn't over-engineer it more than I see here. I can't dig in enough to volunteer an alternative solution.